torch/csrc/utils/cpp_stacktraces.cpp

```diff
@@ -47,9 +47,31 @@ bool get_cpp_stacktraces_enabled() {
   return enabled;
 }
 
-bool get_disable_addr2line() {
-  static bool disabled = compute_disable_addr2line();
-  return disabled;
+static torch::unwind::Mode compute_symbolize_mode() {
+  auto envar_c = std::getenv("TORCH_SYMBOLIZE_MODE");
+  if (envar_c) {
+    std::string envar = envar_c;
+    if (envar == "dladdr") {
+      return unwind::Mode::dladdr;
+    } else if (envar == "addr2line") {
+      return unwind::Mode::addr2line;
+    } else if (envar == "fast") {
+      return unwind::Mode::fast;
+    } else {
+      TORCH_CHECK(
+          false,
+          "expected {dladdr, addr2line, fast} for TORCH_SYMBOLIZE_MODE, got ",
+          envar);
+    }
+  } else {
+    return compute_disable_addr2line() ? unwind::Mode::dladdr
```

Comment:
i assume this is the status quo today, we default to dladdr only if we set the env for addr2line disablement

Reply:
yeah, and once this lands we can delete DISABLE_ADDR2LINE in favor of just setting the mode. Hopefully we can enable the fast version by default and get rid of the flags.